### DISCUSSION TITLE
Is it possible to use groq API

### DISCUSSION THREAD

**Discussion by aborruso** [gptme/gptme#224] [Q&A]
Hi,
I have tried to set [groq api key](https://console.groq.com/docs/openai)

```
OPENAI_API_KEY = "gsk_CIhrph70PrymmsZU1xxxxxxxxx"
OPENAI_BASE_URL = "https://api.groq.com/openai/v1"
```

But I have 

```
raise self._make_status_error_from_response(err.response) from None
openai.AuthenticationError: Error code: 401 - {'error': {'message': 'Incorrect API key provided: gsk_CIhr********************************************RHuz. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}
```

Am I setting it in a wrong way?

Thank you very much

**Comment by ErikBjare** [comment]
I just added proper groq support in 4299cd0, ~~but no new release yet so you'll have to install from source for now.~~

Use it with:

```sh
gptme -m groq/llama-3.1-70b-versatile hello
```

It's possible it didn't work for you as expected due to you using v0.20.0 which did not include the changes https://github.com/ErikBjare/gptme/commit/30e3f01 which are referenced in the docs.

I just released v0.21.0 to solve this.

**Reply by aborruso** [reply]
Hi,
thank you very much.

I have set in my config  `GROQ_API_KEY = "gsk_CIhrpxxxxxxoXQLVufJ07jBXPEIRHuz"`.

I want it as default. If I run simply `gptme` I have

```
No API key set for OpenAI, Anthropic, or OpenRouter.
You can get one at:
 - OpenAI: https://platform.openai.com/account/api-keys
 - Anthropic: https://console.anthropic.com/settings/keys
 - OpenRouter: https://openrouter.ai/settings/keys

Your OpenAI, Anthropic, or OpenRouter API key:
```

How to set groq as default?

Thank you

**Reply by aborruso** [reply]
If I run

```
gptme -m groq/llama-3.1-70b-versatile hello
```

I have

![image](https://github.com/user-attachments/assets/cd4e375c-a6dc-4f3f-95fb-c75fc244e29a)

If then I ask `a name for a cat`

I have

```
File "/home/aborruso/.local/pipx/venvs/gptme/lib/python3.11/site-packages/openai/_base_client.py", line 1058, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.BadRequestError: Error code: 400 - {'error': {'message': "'messages.2' : for 'role:assistant' the following must be satisfied[('messages.2.content' : value must be a string)]", 'type': 'invalid_request_error'}}
```

Thank you

**Comment by TimeToBuildBob** [comment]
Late follow-up, but this thread is still a top hit for "gptme + Groq" and the advice above has aged out, so here is the current state (gptme 0.32.1).

**Your original 401 is now documented behaviour, not a misconfiguration.** Pointing `OPENAI_BASE_URL` at `https://api.groq.com/openai/v1` with `OPENAI_API_KEY` will always 401 — that is why the error message talked about `platform.openai.com`. gptme routes Groq through its own provider rather than the OpenAI-compatible env pair:

```sh
export GROQ_API_KEY="gsk_..."
gptme "hello" -m groq/llama-3.3-70b-versatile
```

**Two things in the answer above are stale:**

- `groq/llama-3.1-70b-versatile` no longer resolves — Groq retired that model. gptme's registry now carries `llama-3.3-70b-versatile` (128k context) as the Groq entry.
- "install from source" is no longer needed. Groq support has been in released gptme for a long time: `pipx install gptme` (or `uv tool install gptme`).

**Also new since this thread:** you can store the key without putting it in shell history or a dotfile —

```sh
gptme '/account setup groq'
```

which writes to `~/.config/gptme/credentials.toml`. The `[env]` section of `~/.config/gptme/config.toml` still works too.

Full provider docs: https://gptme.org/docs/providers.html
